freeipa/src/main/java/com/sequenceiq/freeipa/controller/UserV1Controller.java

```diff
@@ -107,6 +109,17 @@ public SyncOperationStatus getSyncOperationStatus(@NotNull String operationId) {
                 operationService.getOperationForAccountIdAndOperationId(accountId, operationId));
     }
 
+    private String determineAccountId(String callingActor, String requestedAccountId) {
+        if (requestedAccountId == null) {
+            return ThreadBasedUserCrnProvider.getAccountId();
+        }
+        String callingActorAccountId = Crn.safeFromString(callingActor).getAccountId();
+        if (!callingActorAccountId.equals(requestedAccountId) && !InternalCrnBuilder.isInternalCrn(callingActor)) {
+            throw new BadRequestException("Only the internal actor can set a different accountId");
```

Comment:
do we really want to propagate that information to customers if somehow they can invoke it with different accountid?

Reply:
I think that permission denied be more appropriate than bad request then. 

The message should just say that the user isn't allowed to make this call. I agree that we don't want to reveal that internal actors exist.

Reply:
We're not exposing this to customers (they won't have an option to set an accountId through public api). They might somehow figure out how to call this endpoint without going through the public api but we don't need to give friendly message in this case. The message as it is right now is more useful to internal developers.

Reply:
I agree that it should be Permission Denied.  I would prefer to have the same exception message as thrown by the Scrutinizer (see validateActorCrnInAccountOrInternal). You can put any other information you think is useful in a log message.

Reply:
Thanks, I've made the changes.